ts/Series/Map/MapSeries.ts

```diff
@@ -1288,7 +1286,8 @@ addEvent(MapSeries, 'update', function (
 
     if (options.mapData) {
         // Save the current (new) option that will be used in series.setData
-        series.options.mapData = options.mapData;
+        series.options.mapData =
+            merge(series.userOptions.mapData, options.mapData);
```

Comment:
In the case ([demo](http://jsfiddle.net/BlackLabel/q201cfgy/)) of `mapData` being updated to a new array the merge I have suggested fails because when the old option is `undefined` (because `map` is set on chart level) - this forces an object out of the new array and fails in `forEach`.

How about `merge(true, series.options, { mapData: options.mapData });` instead?